### DISCUSSION TITLE
Toot templates

### DISCUSSION THREAD

**Discussion by eiland-asbl** [Automattic/wordpress-activitypub#596] [Ideas]
For different content types, a different toot template might be suitable.

For example, i like to toot links to news publications, which include the url of the news paper - this is a separate post-type, but i also toot my own posts, where the url of my wordpress install would be the link to share

**Comment by cybeardjm** [comment]
Hi, you should have a look at the IndieBlocks plugin by Jan Boddez that adds the "note" and "like" post-types to a WP blog, aka "'short-form' post types".

**Comment by eiland-asbl** [comment]
interesting, but how would i connect the indieblock to the toot templates?

**Comment by cybeardjm** [comment]
Don't know what you mean with "toot templates". How you send to the Fediverse from WP depends on what you select in Settings -> ActivityPub -> Settings -> Activities, for Post Content & Activity-Object-Type (and enabling specific post-types in Supported post types)

**Comment by eiland-asbl** [comment]
Indeed, but there is now only one template for all post-types, whereas i
argue for possible different template for different post-types.

**Reply by pfefferle** [reply]
I have a PR to do that with a filter, because I am not sure how a UI could look like.

**Reply by pfefferle** [reply]
https://github.com/Automattic/wordpress-activitypub/pull/597

**Reply by eiland-asbl** [reply]
Great! How would the filter switch per content-type?
